### DISCUSSION TITLE
Rapid development workflow? SHA512, multi-branch, fork

### DISCUSSION THREAD

**Discussion by SamuelMarks** [microsoft/vcpkg#23672] [Q&A]
My general workflow is:
```sh
$ cd "$google_cloud_c_repo" && git log -1 # get the hash of latest
$ cd "$my_vcpkg_fork" && git checkout google-cloud-c
$ # to bring in all the ports I'm working on so that no errors occur
$ for b in libfswatch randomstr dmon libcurl-simple-https google-cloud-bucket-c cauthflow; do
    git checkout "$b" ports/"$b";
  done ; git reset
$ nano ports/google-cloud-c/portfile.cmake # set `REF` to `git log -1` and `SHA512` to `0`
$ ./vcpkg remove google-cloud-c
$ ./vcpkg install google-cloud-c  # take error to get actual SHA512 hash it wants
$ nano ports/google-cloud-c/portfile.cmake  # set `SHA512` hash
$ ./vcpkg install google-cloud-c  # if no error, then:
$ git commit -S -am '[ports/google-cloud-c/portfile.cmake] <message from `git log -1` on first line>'
$ git clean -fd  # to remove unused ports `git checkout`ed above
$ ./vcpkg x-add-version --all --overwrite-version  # I know this is just if it's ready, but I still want CI kickoff
$ git commit -S -am '`./vcpkg x-add-version --all --overwrite-version`'
$ git push
$ git checkout project0  # branch where all my custom ports lie
$ git merge google-cloud-c  # `:wq`
$ git push
```

I could script these 18 lines into a simple bash or other script… but I'm confident my workflow is too verbose regardless, and there's a more efficient workflow that you guys would use internally [and externally].

Thanks for all suggestions

**Comment by autoantwort** [comment]
Maybe use `--head` mode

**Reply by dg0yt** [reply]
This is the answer if the portfiles and manifests don't need updates.

**Comment by Osyotr** [comment]
Maybe use overlay-ports feature?

**Reply by dg0yt** [reply]
This is the answer if you need to integrate updates to portfiles and manifests. This can include updates of upstream revisions.

**Comment by SamuelMarks** [comment]
@dg0yt @autoantwort @Osyotr Thanks for your replies. As a confounder, say I am developing 7 ports concurrently; some interdependent; and I also want other engineer(s) working on the project to get the latest versions of all 7 of my ports.

So you're thinking I work from just my `project0` branch of my vcpkg port, and setup a portfile like:
```cmake
vcpkg_from_github(
    OUT_SOURCE_PATH SOURCE_PATH
    REPO            "offscale/${PORT}"
    #REF             73e969074e20dbe91b0148ec96024bb7e9a40840
    #SHA512          4b2addbef89516b55ab7f5ab905b9c0faf3a12aff7ad405751b17d577eea8f987d72c19246ed49fe11a148b7a0a862dcfc5e845c0eabea8bcc7d8184e4d1c23d
    HEAD_REF        master
)
```
…and do I need to clear my cache and buildtree or will this suffice? `./vcpkg remove google-cloud-c && ./vcpkg install "$_"`

I'm just trying to get as productive as if I were working in submodules… or better still, as if I had just symlinks setup (but I still want the proper cmake process to run for cross-platform support and install sanity).
